### PR TITLE
Feat/nav logout link

### LINKED ISSUE #1772: add ability to log out under the my profile icon
as a user
with viewer permission or higher
when i am using the application
then i would like to be able to log out from anywhere in the app

similar to other apps, let's add a sign out option under the my profile section of the left menu

### PR DESCRIPTION
**Pull request by alexpaxton**
Closes #1772 

logout-link

Refactored the navbar to be truly composable and support sub-menu items.
Created a logout route `/logout` and a corresponding page component
Replaced the existing logout button with a link to `/logout`

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/Logout.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// Components
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+type Props = WithRouterProps
+
+@ErrorHandling
+export class Logout extends PureComponent<Props> {
+  public async componentDidMount() {
+    this.handleSignOut()
```
Comment (OfTheDelmer) [inline ui/src/Logout.tsx]: does this need to await?

File: ui/src/Logout.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  public render() {
+    return null
+  }
+
+  private handleSignOut = async () => {
+    await logout()
```
Comment (OfTheDelmer) [inline ui/src/Logout.tsx]: What happens if they are offline?

File: ui/src/pageLayout/components/NavMenuItem.tsx [hunk 3]
```diff
@@ -25,12 +33,16 @@ const NavMenuItem: SFC<Props> = ({
...
     <div className={classnames('nav--item', {active: isActive})}>
       <Link className="nav--item-icon" to={link}>
-        {children}
+        <span className={`icon sidebar--icon ${icon}`} />
       </Link>
       <div className="nav--item-menu">
         <Link className="nav--item-header" to={link}>
           {title}
         </Link>
+        {React.Children.map(
```
Comment (OfTheDelmer) [inline ui/src/pageLayout/components/NavMenuItem.tsx]: We could utilize the `Select` component here

```jsx
<Select type={NavMenuSubItem}>
   {this.props.children}
</Select>
```

File: ui/src/Logout.tsx (hunk 2 above)
Comment (alexpaxton) [reply ui/src/Logout.tsx]: They get redirected back to signin no matter what

File: ui/src/pageLayout/components/NavMenuItem.tsx (hunk 3 above)
Comment (alexpaxton) [reply ui/src/pageLayout/components/NavMenuItem.tsx]: Good call!

**Pull request merged**